pilot/pkg/networking/plugin/authz/rbac.go

```diff
@@ -468,6 +481,128 @@ func buildHTTPFilter(service *serviceMetadata, option rbacOption, is11 bool) *ht
 	return out
 }
 
+// convertRbacRulesToFilterConfigV2 is the successor of convertRbacRulesToFilterConfig, which supports
+// converting AuthorizationPolicy.
+func convertRbacRulesToFilterConfigV2(service *serviceMetadata, option rbacOption) *http_config.RBAC {
+	rbac := &policyproto.RBAC{
+		Action:   policyproto.RBAC_ALLOW,
+		Policies: map[string]*policyproto.Policy{},
+	}
+	namespace := service.attributes[attrDestNamespace]
+	// Check to make sure we are not dereferencing null pointers.
+	if option.authzPolicies == nil || option.authzPolicies.NamespaceToAuthorizationConfigV2 == nil {
+		return &http_config.RBAC{Rules: rbac}
+	}
+	if _, present := option.authzPolicies.NamespaceToAuthorizationConfigV2[namespace]; !present {
+		return &http_config.RBAC{Rules: rbac}
+	}
+
+	// Get all AuthorizationPolicy Istio config from this namespace.
+	allAuthzPolicies := option.authzPolicies.NamespaceToAuthorizationConfigV2[namespace].AuthzPolicies
+	// If there is no AuthorizationPolicy in this namespace, try converting using the old converting function.
+	if len(allAuthzPolicies) == 0 {
+		return convertRbacRulesToFilterConfig(service, option)
+	} else if option.globalPermissiveMode {
+		// Permissive Mode will be ignored.
+		// TODO(pitlv2109): Handle permissive mode in the future.
+		return &http_config.RBAC{Rules: rbac}
+	}
+	// Get all key-value pairs of AuthorizationPolicy name to list of Subjects/ServiceRoleBindings in this namespace.
+	authzPolicyToSubjects := option.authzPolicies.AuthzPolicyToAllowSubjects(namespace)
+	for _, authzPolicy := range allAuthzPolicies {
+		// If a WorkloadSelector is used in the AuthorizationPolicy config and does not match this service, skip this
+		// AuthorizationPolicy.
+		if authzPolicy.Policy.WorkloadSelector != nil && !service.isServiceMatchedWorkload(authzPolicy.Policy.WorkloadSelector.Labels) {
```

Comment:
Please double check the labels from the service is the right one to use. If I remember correctly, the labels in service is not the labels for workload.
To test this, create different label on a service and workload and print its values here.

Reply:
As we discussed, since the Labels field in ServiceInstance are workload labels, no fix is needed for this.